crates/re_renderer/src/allocator/cpu_write_gpu_read_belt.rs

```diff
@@ -119,21 +119,30 @@ where
         elements: impl Iterator<Item = T>,
     ) -> Result<usize, CpuWriteGpuReadError> {
         re_tracing::profile_function!();
-        let num_written_before = self.num_written();
 
-        for element in elements {
-            if self.unwritten_element_range.start >= self.unwritten_element_range.end {
-                return Err(CpuWriteGpuReadError::BufferFull {
-                    buffer_element_capacity: self.capacity(),
-                });
+        // TODO(emilk): optimize the extend function.
+        // Right now it is 3-4x faster to collect to a vec first, which is crazy.
```

Comment:
oh wow, that's so sad. I wonder what aspect of this method is this and if that's true with all iterators or only some.

Reply:
I'm guessing all the added bounds check